learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+# Multitenancy and tenant tokens
+
+In this guide you'll first learn [what multitenancy is](#what-is-multitenancy) and [how tenant tokens help managing complex applications and sensitive data](#what-are-tenant-tokens-and-how-do-they-solve-multitenancy-in-meilisearch). Then, you'll see [how to generate and use tokens](#generating-tenant-tokens-with-an-sdk), whether with an official SDK or [otherwise](#generating-tenant-tokens-without-a-meilisearch-sdk). The guide ends with a quick explanation of [the main tenant token settings](#tenant-token-payload-reference).
+
+## What is multitenancy?
+
+In software development, multitenancy means that multiple users with different interests—also called tenants—share the same computing resources. Proper multitenancy is crucial in cloud computing services such as [Digital Ocean's Droplets](https://www.digitalocean.com/products/droplets) and [Amazon's AWS](https://aws.amazon.com/).
+
+If your Meilisearch application stores sensitive data like medical records in the same index, it is very important to make sure users can only search through their own documents.
+
+### What are tenant tokens and how do they solve multitenancy in Meilisearch?
+
+Tenant tokens are small packages of encrypted data presenting proof a user can access a certain index. They contain not only security credentials, but also instructions on which documents within that index the user is allowed to see.
+
+To use tokens in Meilisearch, you only need to have a system for token generation in place. The quickest method to generate tenant tokens is [using one of our official SDKs](#generating-tenant-tokens-with-an-sdk). It is also possible to [generate a token from scratch](#generating-tenant-tokens-without-a-meilisearch-sdk).
+
+Tenant tokens do not require you to configure any specific [instance options](/learn/configuration/instance_options.md) or [index settings](/learn/configuration/settings.md). They are also meant to be short-lived—Meilisearch does not store nor keeps track of generated tokens.
+
+## Generating tenant tokens with an SDK
+
+If you are developing an application that allows patients to search through medical records, it is crucial that each user can only see their own records and not anyone else's.
+
+The code in this example imports the SDK, creates a filter based on the current user's ID, and feeds that data into the SDK's `generateTenantToken` function. Once the token is generated, it is stored in the `token` variable:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_generate_sdk_1">
+const client = new Meilisearch({host: 'meilisearch'});
+
+const apiKey = '[an_api_key]';
+const currentUserID = '[placeholder_user_id]';
+
+const searchRules = {
+  'patient_medical_records': {
+    'filter': `user_id = ${currentUserID}`
+  }
+};
+
+const expiryDate = 1645461882;
+
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
```

Comment:
The code sample will look like this: 

```js
const client = new Meilisearch({ host: 'meilisearch' });

const apiKey = '[an_api_key]';
const currentUserID = '[placeholder_user_id]';
const expiresAt = new Date('2030-04-01');
const searchRules = {
  'patient_medical_records': {
    'filter': `user_id = ${currentUserID}`
   }
};


const token = client.generateTenantToken(searchRules, {
   expiresAt: expiresAt,
   apiKey: apiKey,
});
```

Reply:
Looks great. Only one comment: we have been moving away from marking placeholder values with `[]` (e.g. `[an_api_key]`). @dichotommy, @maryamsulemani97 do you think we should also enforce this in SDK examples?

Reply:
I think removing `[]` is a good idea as well :) It is a bit confusing. If so should I replace it with `an_api_key` ? (same for user_id)

Reply:
Ok, cool, so let's go with that, then: placeholder values without `[]`. You probably don't need to update it here, as I'm going to replace the hardcoded sample with a `<CodeSample>` component soon.

Thanks!